pennylane/math/matrix_manipulation.py

```diff
@@ -120,3 +123,91 @@ def expand_matrix(base_matrix, wires, wire_order=None):
+    return kron(
+        kron(eye(2 ** (j - 1)), swap_mat), eye(2 ** (n - (j + 1))), format=format
+    )  # (j - 1) + 2 + (n - (j+1)) = n
+
+
+def _sparse_swap_mat(i, j, n, format="csr"):
+    """Helper function which generates the sparse matrix of SWAP
+    for qubits: i <--> j with final shape (2**n, 2**n)."""
+    assert i < n and j < n
+    if i == j:
+        return eye(2**n, format=format)
+
+    (small_i, big_j) = (i, j) if i < j else (j, i)
+    store_swaps = [
+        _local_sparse_swap_mat(index, n, format=format) for index in range(small_i, big_j)
+    ]
+
+    res = eye(2**n, format=format)
+    for mat in store_swaps:  # swap i --> j
+        res @= mat
+
+    for mat in store_swaps[-2::-1]:  # bring j --> old_i
+        res @= mat
+
+    return res
+
+
+def sparse_expand_matrix(base_matrix, wires, wire_order=None, format="csr"):
+    """Re-express a sparse base matrix acting on a subspace defined by a set of wire labels
+    according to a global wire order.
+
+    Args:
+        base_matrix (scipy.sparse.spmatrix): base matrix to expand
+        wires (Iterable): wires determining the subspace that base matrix acts on; a base matrix of
+            dimension :math:`2^n` acts on a subspace of :math:`n` wires
+        wire_order (Iterable): global wire order, which has to contain all wire labels in ``wires``, but can also
+            contain additional labels
+        format (str): string representing the preferred scipy sparse matrix format to cast the expanded
+            matrix too
+
+    Returns:
+        tensor_like: expanded matrix
+    """
+
+    if (wire_order is None) or (wire_order == wires):
+        return base_matrix
+
+    interface = qml.math.get_interface(base_matrix)  # pylint: disable=protected-access
+    if interface != "scipy" and not issparse(base_matrix):
+        raise ValueError("base_matrix must be a scipy sparse matrix")
+
+    n_wires = len(wires)
+    n_total_wires = len(wire_order)
+
+    expanded_wires = copy.copy(wires)
+    for wire in wire_order:
+        if wire not in wires:
+            expanded_wires.append(wire)
+
+    num_missing_wires = n_total_wires - n_wires
+    if num_missing_wires > 0:
+        expanded_matrix = kron(
+          base_matrix, eye(2**num_missing_wires, format=format), format=format
+        )  # added missing wires at the end
+    else:
+        expanded_matrix = copy.copy(base_matrix)
+
+    U = eye(2**n_total_wires, format=format)
+    while not expanded_wires == wire_order:
+        for i in range(n_total_wires):
+            if expanded_wires[i] != wire_order[i]:
+                j = wire_order.index(expanded_wires[i])
+                U = U @ _sparse_swap_mat(i, j, n_total_wires, format=format)
+
+                expanded_wires[i], expanded_wires[j] = expanded_wires[j], expanded_wires[i]
+                break
+
+    expanded_matrix = csr_matrix(U.T) @ expanded_matrix @ U
```

Comment:
```suggestion
    expanded_matrix = (U.T).asformat(format) @ expanded_matrix @ U
```

Reply:
Added, but I casted the final expanded matrix 👍🏼

Reply:
Yeah. That's why I asked if you were really getting an error or the wrong sparse format returned by the function.